examples/multimodal/rag.ts

```diff
@@ -38,17 +39,18 @@ async function main() {
     llm,
     chunkSize: 512,
     chunkOverlap: 20,
-    callbackManager,
   });
   const index = await createIndex(serviceContext);
 
   const queryEngine = index.asQueryEngine({
     responseSynthesizer: new MultiModalResponseSynthesizer({ serviceContext }),
     retriever: index.asRetriever({ similarityTopK: 3, imageSimilarityTopK: 1 }),
   });
-  const result = await queryEngine.query({
-    query: "Tell me more about Vincent van Gogh's famous paintings",
-  });
+  const result = await runWithCallbackManager(callbackManager, () =>
```

Comment:
Is this runWithCallbackManager the only way to get this functionality?

I think for the most natural DevEx the user should do something like registerCallbackManager one, and then everything should just automatically flow to that.

Reply:
Almost like a `window.addEventListener` (actually, that makes me wonder if that would be an acceptable interface to hook into.

Reply:
As mentioned in the description, you could do `Settings.callbackManager = new CallbackManager` if you want

Reply:
`runWith` is more like fine-grained tracking situation.

For example for tracking LLM cost for a single call, you call use `runWithCallbackManager(myManager, () => llm.query())`, then you could get usage for this single call.

But you want to do some global injection, I think we should suggest user to use Settings API

Reply:
OK let's make that the default in our examples. I'm guessing most beginner users will only need one CallbackManager for one specific thing (i.e. logging or Weights and Biases). Btw, should we make that an array? Or is there an easy way to chain CallbackManagers?

Reply:
I'm making it chainable now; see https://github.com/run-llama/LlamaIndexTS/pull/675#discussion_r1544860405

I'm imaging that you can do like 

```
cb
  .addHandler('chat-start', () => {})
  .addHandler('chat-start', () => {})
  .addHandler('chat-end', () => {})
```